src/webresource-handler/handlers/S3Handler.ts

```diff
@@ -122,6 +131,145 @@ export class S3Handler implements WebResourceHandler {
 		return webResource;
 	}
 
+	public async getUpload(
+		uploadParameters: MultipartUploadParameters,
+		metadata: MultipartUploadBody,
+	): Promise<MultipartUploadResponse> {
+		const fileKey = this.getFileKey(
+			uploadParameters.fieldName,
+			metadata.filename,
+		);
+
+		const createMultiPartResponse = await this.client.send(
+			new CreateMultipartUploadCommand({
+				Bucket: this.bucket,
+				Key: fileKey,
+				ContentType: metadata.content_type,
+			}),
+		);
+
+		if (createMultiPartResponse.UploadId == null) {
+			throw new WebResourceError('Failed to create multipart upload.');
+		}
+
+		const [token, uploadUrls] = await Promise.all([
+			this.generateMetadataJwt(
+				fileKey,
+				createMultiPartResponse.UploadId,
+				uploadParameters,
+				metadata,
+			),
+			this.getPartUploadUrls(
+				fileKey,
+				createMultiPartResponse.UploadId,
+				metadata,
+			),
+		]);
+		return { token, uploadUrls };
+	}
+
+	public async decodeUploadToken(
+		token: string,
+	): Promise<MultipartUploadTokenPayload> {
+		return jsonwebtoken.verify(
+			token,
+			this.jwtSigningKey,
+		) as MultipartUploadTokenPayload;
```

Comment:
Result of `verify` is `string | JwtPayload` where JwtPayload is basically `any` so we need to cast as this does not satisfies MultipartUploadTokenPayload

Reply:
How do you know it’s not a string?

Reply:
@dfunckt The only place a valid token should be generate is above, where we generate from an object that has the correct `MultipartUploadTokenPayload` type. If verify works (e.g. the token was signed with a valid key) then it must have come from a MultipartUploadTokenPayload.

Reply:
So, because of signing, right?

Can you please leave a short comment in the code to avoid causing a pause to think about it when someone reads that bit in the future?